uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/Service/LeaveRequestCommentTest.php

```diff
@@ -0,0 +1,321 @@
+<?php
+
+use CRM_HRCore_Test_Fabricator_Contact as ContactFabricator;
+use CRM_HRComments_Test_Fabricator_Comment as CommentFabricator;
+use CRM_HRComments_BAO_Comment as Comment;
+use CRM_HRLeaveAndAbsences_Test_Fabricator_LeaveRequest as LeaveRequestFabricator;
+
+/**
+ * Class CRM_HRLeaveAndAbsences_Service_LeaveRequestCommentTest
+ *
+ * @group headless
+ */
+class CRM_HRLeaveAndAbsences_Service_LeaveRequestCommentTest extends BaseHeadlessTest {
+
+  use CRM_HRLeaveAndAbsences_LeaveManagerHelpersTrait;
+
+  public function setUp() {
+    CRM_Core_DAO::executeQuery("SET foreign_key_checks = 0;");
+  }
+
+  public function tearDown() {
+    CRM_Core_DAO::executeQuery("SET foreign_key_checks = 1;");
+  }
+
+  public function testAddCanCreateCommentForLeaveRequest() {
+    $params = [
+      'leave_request_id' => 1,
+      'text' => 'Random Commenter',
+      'contact_id' => 1,
+      'sequential' => 1
+    ];
+
+    $service = new CRM_HRLeaveAndAbsences_Service_LeaveRequestComment($params);
```

Comment:
If you remove `$params` from the constructor, then you'll be able to instantiate the service on the `setUp()` and avoid having to create it on every test